spritesheet-manager.js

```diff
@@ -29,13 +33,35 @@ class SpritesheetManager {
     }
     return spritesheet;
   }
+
   async getSpriteSheetForAppUrlAsync(appUrl, opts) {
-    let spritesheet = this.spritesheetCache.get(appUrl);
-    if (!spritesheet) {
-      spritesheet = await this.getSpriteSheetForAppUrlInternal([appUrl, opts]);
-      this.spritesheetCache.set(appUrl, spritesheet);
+    const spritesheet = this.spritesheetCache.get(appUrl);
+
+    if (spritesheet) return spritesheet;

```

Comment:
In the rest of the codebase, early returns are not preferred (instead we nest ifs), and curlies are non-optional.

Reply:
I updated the code and changed the logic for loading assets. Now it creates predefined number of loaders and poll them to check whether they are idle or not. If they are then assign them tasks to load the asset. If there is no loader which is idle then then loading of the assets wait in the queue.

I also changed my code style of early return to nested ifs.

Worked on the race conditions and try to mitigate them.